lib/Museum.php

```diff
@@ -0,0 +1,26 @@
+<?
+
+class Museum extends PropertiesBase{
+	public $MuseumId;
+	public $Name;
+	public $Domain;
+
+	/**
+	 * @return array<Museum>
+	 */
+	public static function GetAll(): array{
+		return Db::Query('
+			SELECT *
+			FROM Museums
+			ORDER BY Name', [], 'Museum');
+	}
+
+	public static function FindMatch(string $url): ?Museum{
+		foreach(Museum::GetAll() as $museum){
+			if(stristr($url, $museum->Domain)){
+				return $museum;
+			}
+		}
+		return null;
```

Comment:
Minor suggestion: you could do the search in the database rather than fetching all possibilities and searching them in PHP. It's a very small set though so really comes down to personal preference.

```suggestion
		$result = Db::Query('
			SELECT *
			FROM Museums
			WHERE ? LIKE CONCAT("%", Domain, "%")
			LIMIT 1;
		', [$url], 'Museum');

		return $result[0] ?? null;
```

Reply:
I like it. Updated. Thanks for proposing another way to do it.

My old `GetAll()` approach may have been better suited to using the APCu (or even just relying on the DB query cache), but this is less code.